### PR TITLE
feat(RDS): add rds mysql account params and rewrite for all

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:

1.   rewrite for all
2.  add params description and hosts

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add rds mysql account params and rewrite for all
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/rds/' TESTARGS='-run TestAccMysqlAccount_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds/ -v -run TestAccMysqlAccount_basic -timeout 360m -parallel 4
=== RUN   TestAccMysqlAccount_basic
--- PASS: TestAccMysqlAccount_basic (561.79s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       561.834s
```

### CODE REVIEW

File: huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go [hunk 1]
```diff
@@ -91,85 +114,144 @@ func resourceRdsAccountCreate(ctx context.Context, d *schema.ResourceData, meta
...
+	getMysqlAccountBasePath = strings.ReplaceAll(getMysqlAccountBasePath, "{instance_id}", instanceId)
 
-	// items on every page, [1, 100]
-	limit := int32(100)
-	// List all db users
-	request := &model.ListDbUsersRequest{
-		InstanceId: instanceId,
-		Limit:      limit,
-		Page:       int32(1),
+	page := 1
```
Comment (hmbig2) [inline huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go]: maybe you can declare a query parameter: `page`, then will auto-generate those code

File: huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go [hunk 2]
```diff
@@ -91,85 +114,144 @@ func resourceRdsAccountCreate(ctx context.Context, d *schema.ResourceData, meta
...
+		}
+
+		getMysqlAccountRespBody, err := utils.FlattenResponse(getMysqlAccountResp)
 		if err != nil {
-			return common.CheckDeletedDiag(d, err, "error listing RDS db users")
+			return diag.FromErr(err)
 		}
-		users := *response.Users
+
+		users := utils.PathSearch("users", getMysqlAccountRespBody, make([]interface{}, 0)).([]interface{})
```
Comment (hmbig2) [inline huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go]: add parameter to query api:  x-ref-filter: users[?name=='%s']|[0]

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/rds/' TESTARGS='-run TestAccMysqlAccount_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds/ -v -run TestAccMysqlAccount_basic -timeout 360m -parallel 4
=== RUN   TestAccMysqlAccount_basic
--- PASS: TestAccMysqlAccount_basic (600.89s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       600.985s

File: huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go (hunk 1 above)
Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go [hunk 3]
```diff
@@ -40,182 +44,312 @@ func ResourceRdsAccount() *schema.Resource {
...
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Specifies the IP addresses that are allowed to access your DB instance.`,
 			},
 			"password": {
-				Type:      schema.TypeString,
-				Required:  true,
-				Sensitive: true,
+				Type:        schema.TypeString,
```
Comment (ShiChangkuo) [inline huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go]: 1. **Sensitive** is missing
2. should define the `Required` parameter in head of `Optional` parameters

File: huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go [hunk 4]
```diff
@@ -40,182 +44,312 @@ func ResourceRdsAccount() *schema.Resource {
...
-		InstanceId: instanceId,
-		Body: &model.UserForCreation{
-			Name:     dbUser,
-			Password: d.Get("password").(string),
-		},
+	createMysqlAccountOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
 	}
+	createMysqlAccountOpt.JSONBody = utils.RemoveNil(buildCreateMysqlAccountBodyParams(d))
+	log.Printf("[DEBUG] Create RDS Mysql account options: %#v", createMysqlAccountOpt)
```
Comment (ShiChangkuo) [inline huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go:105]: it's not advised to log the password in **Create** and **Update** options

File: huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go:105]: fixed

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/rds/' TESTARGS='-run TestAccMysqlAccount_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds/ -v -run TestAccMysqlAccount_basic -timeout 360m -parallel 4
=== RUN   TestAccMysqlAccount_basic
--- PASS: TestAccMysqlAccount_basic (551.69s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       1066.028s

**Pull request merged**